internal/mage/util/engine.go

```diff
@@ -0,0 +1,327 @@
+package util
```

Comment:
This is a nit but it seems like it'll be a bit confusing to remember what's handled in `internal/mage/engine.go` and what's handled in `internal/mage/util/engine.go`. Would it be easy to update to something like:
1. `internal/mage/engine/*` is added and handles most of what's currently in `internal/mage/util/engine.go` and `internal/mage/engine.go` 
2. For the engine publish step, which requires the SDK code and thus needs to avoid import loops, just put that in `internal/mage/publish.go` or something like that?

If this is harder than it sounds or there's disagreements, then no worries at all, this is far from a blocking concern

Reply:
Yeah, that makes some sense, @sipsma. 

It isn't quite clear, but the distinction I'll note is that `internal/mage/engine.go` is for the mage tasks, and `internal/mage/util/engine.go` is for utilities related to dev engine. 

I'm nearing the end of my energy for today, which has me hesitant to dig into the rest of the refactor that shifting this would require. I'll take another look on Monday, and we can make a decision then.